src/components/GettingStartedModal/index.tsx

```diff
@@ -0,0 +1,180 @@
+  // partial means any field can be undefined
+  wizardState: Partial<WizardState>;
+  setWizardState: Dispatch<SetStateAction<Partial<WizardState>>>;
+
+  // these functions let us navigate through the steps
+  // undefiend if on first step
+  goToPreviousStep?: () => void;
+  goToNextStep?: () => void; 
+  exitWizard?: () => void;
+}
+
+export interface Step {
+  title: string;
+  Component: (props: StepProps) => ReactElement;
+  /**
+   * This function says how we should actually update user state.
+   */
+  commitState: (state: WizardState, interactors: UserInteractors) => void;
+}
+
+/**
+ * This is the list of steps for the getting started modal.
+ *
+ */
+const steps: Step[] = [Preamble, GroupRegistrationStep, PhoneticsStep, ChooseSetStep];
+
+export function GettingStartedModal() {
+  const userStateContext = useUserStateContext();
+
+  // keep track of which step of workflow we are on (start on first step)
+  const [stepNumber, setStepNumber] = useState(0);
+  // keep track of what data the user has filled out 
+  const [wizardState, setWizardState] = useState<Partial<WizardState>>({ });
+
+  const [fullState, setFullState] = useState<WizardState | null>(null);
+  //const [fullState, setFullState] = useState<WizardState>({groupId: '', phoneticsPreference: null, collectionId: '', email: '', whereFound: ''});
+  
+  function exitWizard(){
+    const { groupId, email, whereFound } = wizardState;
+
+    // check if all required fields have been filled out
+    if (groupId && email && whereFound) {
+      // if all fields are filled out, update user state and exit the modal
+      const fullState: WizardState = {
+        groupId,
+        email,
+        whereFound,
+      };
+      steps.forEach((step) => step.commitState(fullState, userStateContext));
+      setFullState(fullState);
+      // TODO: perform any necessary cleanup or finalization
+    } else {
+      // if any required fields are missing, show an error message or prevent the user from exiting
+      alert("Please fill out all required fields.");
+      // alternatively, you can disable the exit button until all required fields are filled out
+    }
+  }
+
+  /**
+   * Take all the data the user input and run actions for each step using it
+   */
+  function exitWizardAtEnd() {
+    // any part of wizard state could be undefined, so unpack it all
+    const { groupId, email, phoneticsPreference, collectionId, whereFound} = wizardState;
+    // add checks here to make sure fields are defined
+    if (groupId !== undefined && email !== undefined && phoneticsPreference !== undefined && collectionId != undefined && whereFound !== undefined) {
+      // reassemble state here (without Partial<>)
+      const fullState: WizardState = { groupId, email, phoneticsPreference, collectionId, whereFound};
+      // dispatch the actions for each step
+      steps.forEach((step) => step.commitState(fullState, userStateContext));
+    }
+  }
```

Comment:
how is this different from `exitWizard`? When does each get called?

Reply:
One exits upon filling out all required fields, the other ends after advanced features have been filled out. I have updated exitWizard to exitWizardNoAdvanced to make this more clear.